addons/account/models/account_code_mapping.py

```diff
@@ -72,3 +75,30 @@ def _compute_code(self):
     def _inverse_code(self):
         for record in self:
             record.account_id.with_company(record.company_id._origin).code = record.code
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        @contextmanager
+        def unprotecting(env, fields, records):
+            protected = env._protected
+            try:
+                protected.pushmap()
+
+                for field in fields:
+                    ids = protected.get(field, set())
+                    protected[field] = ids.difference(records.ids)
+                yield
+
+            finally:
+                protected.popmap()
+
+        code_field = self.pool['account.account'].code
+        for vals in vals_list:
+            if any(field_name not in vals for field_name in ('account_id', 'company_id', 'code')):
+                raise UserError(self.env._("When modifying a code mapping for an account, you must specify both the company and the code for that company."))
+            account = self.env['account.account'].browse(vals['account_id'])
+
+            # When the user writes on `account.code_mapping_ids`, they want to modify the code, so we unprotect it temporarily here.
+            with unprotecting(self.env, [code_field], account):
+                account.with_company(vals['company_id']).code = vals['code']
```

Comment:
But even if it is protected, you are writing on the code explicitly. The protection only does something against recomputations AFAIK.

Shouldn't you write on `code_store` directly instead?

Reply:
I thought the protected prevents the field to be written on. In this case, it was needed because if the user provides both a value for `code` and `code_mapping_ids` then the `code` is written first (which affects only the active company) but is then protected, and the protection prevents it from being modified not just in the active company but also for all other companies...

We can also write on `code_store` directly but then that would bypass any constraints on `code` that might be done in other modules (e.g. in the latams).

Reply:
Writing on `code_store` doesn't trigger the constraints on code. Is it because the dependency isn't checked? or because at that point `code` is protected? I'm not sure.

And that's the case even if we make code dependent on code_store (which, by the way, I don't know how I didn't spot that missing depends earlier :laughing:).

We can't trigger the validation manually in `account.code.mapping.create` either since code is protected at that point and therefore its old value will be used in the constraints. 

My solution: ensure that we only pass values either for `code` or for `code_mapping_ids` in the create and write. That way, `code` is never protected when we need to write on `code_mapping_ids`.